pkg/printers/internalversion/printers.go

```diff
@@ -1193,14 +1205,19 @@ func getNodeExternalIP(node *api.Node) string {
 
 // findNodeRole returns the role of a given node, or "" if none found.
 // The role is determined by looking in order for:
+// * a node-role.kubernetes.io/master label
 // * a kubernetes.io/role label
 // * a kubeadm.alpha.kubernetes.io/role label
-// If no role is found, ("", nil) is returned
 func findNodeRole(node *api.Node) string {
```

Comment:
this seems like exactly the sort of special-casing we should not do, given the discussion in https://github.com/kubernetes/kubernetes/pull/39112 about nodes being able to have multiple roles. Wasn't the purpose of the selected label pattern to allow arbitrary roles, and multiple roles per node?